pyccel/ast/internals.py

```diff
@@ -173,6 +179,18 @@ def step(self):
         """
         return self._step
 
+    @property
+    def slice_type(self):
+        """ The type of the slice (Range or Element)
+        Range   <=> [:]
+        Element <=> [3]
+        """
+        return self._slice_type
+
+    @slice_type.setter
+    def slice_type(self, value):
+        self._slice_type = value
```

Comment:
Where do you need the setter as well as the constructor argument?

Reply:
https://github.com/pyccel/pyccel/pull/1299/commits/6331896f3ba794243b8f89d73913440e41f7b0f2